include/leveldb/iterator.h

```diff
@@ -64,6 +65,10 @@ class Iterator {
   // REQUIRES: !AtEnd() && !AtStart()
   virtual Slice value() const = 0;
 
+  // Riak specific:  if a database iterator, returns key meta data
+  // REQUIRES: Valid()
+  virtual KeyMetaData & keymetadata() const {return(keymetadata_); };
```

Comment:
Note that keymetadata_ is mutable.  So this is where things got ugly.  Iterators are const everywhere useful.  Expiry creates the need to return not only the value, but also the KeyMetaData.  I considered several alternatives.  The one that created fewer code changes was to keep the const Iterator but give it a mutable storage for KeyMetaData.  Open to alternate construction ideas.